circleci-orb/scripts/rebase-on-target-branch/index.ts

```diff
@@ -1,6 +1,172 @@
+  exec(`git remote add ${remoteName} https://github.com/${owner}/${name}.git`);
+  exec(`git fetch ${remoteName}`);
+  return remoteName;
+}
+
+/** Get the ref and latest shas for the provided sha on a specific remote. */
+function getRefAndShas(sha: string, owner: string, name: string) {
+  const remote = addAndFetchRemote(owner, name);
+  // Get the ref on the remote for the sha provided.
+  const ref = getRefFromBranchList(getBranchListForSha(sha, remote));
+  // Get the latest sha on the discovered remote ref.
+  const latestSha = getShaFromRef(`${remote}/${ref}`);
+
+  return {remote, ref, latestSha, sha};
+}
+
+/** Rebase on the latest commit for the targetted branch. */
+(async () => {
+  const base = getRefAndShas(baseRevision, baseRepoOwner, baseRepoName);
+  const target = getRefAndShas(headRevision, headRepoOwner, headRepoName);
+  const commonAncestorSha = getCommonAncestorSha(base.sha, target.sha);
+
+  // Log known refs and shas
+  console.log(`--------------------------------`);
+  console.log(`    Target Branch:                   ${base.ref}`);
+  console.log(`    Latest Commit for Target Branch: ${target.latestSha}`);
+  console.log(`    Latest Commit for PR:            ${base.latestSha}`);
+  console.log(`    First Common Ancestor SHA:       ${commonAncestorSha}`);
+  console.log(`--------------------------------`);
+  console.log();
+
+  // Get the count of commits between the latest commit from origin and the common ancestor SHA.
+  const commitCount = exec(`git rev-list --count origin/${base.ref}...${commonAncestorSha}`);
+  console.log(`Checking ${commitCount} commits for changes in the CircleCI config file.`);
+
+  // Check if the files changed between the latest commit from origin and the common ancestor SHA
+  // includes the CircleCI config.
+  const circleCIConfigChanged = exec(
+    `git diff --name-only origin/${base.ref} ${commonAncestorSha} -- .circleci/config.yml`,
+  );
+  if (!!circleCIConfigChanged) {
+    throw Error(`
+        CircleCI config on ${base.ref} has been modified since commit
+        ${commonAncestorSha.slice(0, 7)}, which this PR is based on.
+
+        Please rebase the PR on ${base.ref} after fetching from upstream.
+
+        Rebase instructions for PR Author, please run the following commands:
+
+          git fetch upstream ${base.ref};
+          git checkout ${target.ref};
+          git rebase upstream/${base.ref};
+          git push --force-with-lease;
+
+        Angular team members can rebase this PR with the following command:
+          yarn ng-dev pr rebase <pr-number>
+        `);
+  } else {
+    console.log('No change found in the CircleCI config file, continuing.');
+  }
+  console.log();
+
+  // Rebase the PR.
+  exec(`git rebase origin/${base.ref}`);
+  console.log(`Rebased current branch onto ${base.ref}.`);
+})().catch((err: unknown) => {
+  console.log('Failed to rebase on top of target branch.\n');
+  console.error(err);
+  process.exitCode = 1;
+});
```

Comment:
Can you also add the new "inputs" to the Yaml config?

Reply:
I updated the inputs in YAML. However, because most of these environment variables are not part of the pipeline values, we just rely on them always being established in the environment already.